src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorage.cs

```diff
@@ -306,5 +267,51 @@ public override void Initialize(Solution solution)
                 BulkPopulateIds(connection, solution, fetchStringTable);
             }
         }
+
+        internal bool TryOpen()
+        {
+            // Attempt to create a connection to the DB and configure the database, handling locked database failure.
+            try
+            {
+                using (var pooledConnection = GetPooledConnection())
```

Comment:
1. have you validated this works by actually debugging through things?
2. Can we add a test for this?  It should not be hard to do.

Reply:
I validated that this works by adding some logging, installing custom-built Roslyn to a fresh install of VS 15.4.0 and then launching several copies of VS with one solution. Only one instance of VS passed through this, others did not use SQLite persistence.

Reply:
Adding test to verify it should not be hard to do, but there since that shared cache [makes exclusive locking invisible in-process](https://www.sqlite.org/sharedcache.html#shared_cache_locking_model), it may require launching additional processes containing the same Roslyn code being tested. This certainly does not look easy.
I'll do a couple of checks to verify my assumptions about shared cache, maybe I'm mistaken.